src/journal/bsod.c

```diff
@@ -75,13 +78,23 @@ static int acquire_first_emergency_log_message(char **ret) {
         if (r < 0)
                 return log_error_errno(r, "Failed to seek to start of jornal: %m");
 
-        r = sd_journal_next(j);
-        if (r < 0)
-                return log_error_errno(r, "Failed to read next journal entry: %m");
-        if (r == 0) {
-                log_debug("No emergency level entries in the journal");
-                *ret = NULL;
-                return 0;
+        for(;;) {
+                r = sd_journal_next(j);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read next journal entry: %m");
+
+                if (r == 0) {
+                        log_debug("No emergency level entries in the journal");
+                        *ret = NULL;
```

Comment:
do this only just before `return 0` as in other cases it won't return immediately and there might be errors later